Setup Testing Suite

skylerto opened the pull request:
# Summary

This PR contains a testing suite with Mocha and Chai, I also configured grunt with a default test task. 
# Tests

Test cases are located under spec/
## Strategy

The benefit of these testing frameworks are we can inject our own markdown wikis and test the generated code as front end files. Just a testing strategy worth exploring.
## Future

Move to with modularity future?
This PR contains some extra modularity.

with :heart:,
sky

skylerto committed:
Constants and Function

Created constants and function for getting opts.

**FUTURE**
lib/parser.js

skylerto committed:
Testing Suite

Configured grunt and a testing suite. Suite uses mocha and chai. Test can be
executing using `grunt` as test is the default task.

skylerto committed:
Modularize Parser

Built a modular parser, unit tested.

rbren commented on lib/parse.js:
```diff
@@ -0,0 +1,24 @@
+    };
+  }
+  if (!object.directory) {
+    // do some sort of directory finding algorithm.
```
Let's make this a TODO and throw an error for now

rbren commented on lib/parse.js:
```diff
@@ -0,0 +1,24 @@
+  }
+
+  object.marked = object.marked || defaults.marked;
+  //object.marked.gfm = true;
```
Can remove this line

rbren commented on lib/wiki.js:
```diff
@@ -57,3 +56,20 @@ var Wiki = module.exports = function(opts) {
   Marked.setOptions(MARKED_DEFAULTS);
 };
 
+function getOpts(opts){
```
I don't see this function being used?

skylerto replied:
Yikes! This function was replaced by the parser (lib/parse.js)

rbren commented on spec/tests.js:
```diff
@@ -0,0 +1,9 @@
+var chai = require('chai');
```
Is this an auto-generated file?

skylerto replied:
Oh, whoops my test setup script, that's embarrassing! Sorry about that!

bobby-brennan commented:
Thanks! Tests are definitely appreciated.

skylerto committed:
Clean-up and Error

Migrated wiki.js to use the defaults.json file instead of hosting its own
constants.

Cleaned-up generated test files. Also cleaned un-needed options in Gruntfile.

Parse.js now throws an error when no directory is passed in (back to the old way).

bobby-brennan commented:
Thanks!
